### ISSUE TITLE
before_build hook issue

### ISSUE THREAD

**Issue by mohitmayank** [apla/me.apla.cordova.app-preferences#53]
facing a small issue cause after https://github.com/apla/me.apla.cordova.app-preferences/pull/50 
if i remove and re-add android platform getting following error -

```
Error: ENOENT, open 'platforms/android/src/me/apla/cordova/AppPreferencesActivity.java'
    at Error (native)
```
perhaps moving the hook to before_build might help?

I don't add platform directory to version control. I also do customization on plugins and sometimes need to reset platform directory.

**Comment by Treer** [comment]
There are even similiar errors during operations on unrelated plugins — any plugin:
```
> cordova plugin rm cordova-plugin-vibration                                                                      
Error: ENOENT, unlink 'D:\Projects\Mobile\cordova\<appname>\source\platforms\android\res\xml\apppreferences.xml'                                                                                                                 
```
&lt;exasperated whinge&gt;

These issues are bad, nearly every type of cordova operation seems to be broken or get aborted mid-way through without unrolling because of errors that trace back to app-preferences. Even if I start a new project fresh and get to a point where I can get these files to exist, the project has already be borked by things being half installed in unobvious ways (json files not updated etc).

I thought I could work around it - just be careful and find a working order to add platforms and plugins, but after 3 days of whacking moles caused by one plugin, I think the options are:

* Follow [lakani's advice](https://github.com/apla/me.apla.cordova.app-preferences/issues/55#issuecomment-151183791) and switch to an older version (I don't *remember* having any issues like this with app-preferences when I started the project, but perhaps I just didn't have other plugins)

* Remove all app-preferences hooks, or fix the code so none of the hooks throw cordova-operation-aborting errors. And make app-preferences handle "name" as well as "key" instead of crashing cordova silently (the error message app-preferences throws never reaches the console, I had to step through a cordova building operation with a debugger to find why builds weren't working).

(I'm going to try switching to localStorage as I can't spend any more time on this, but I know that's not helpful to app-preferences)
